firestore.rules

```diff
@@ -80,15 +107,45 @@ service cloud.firestore {
     match /authed/{portal} {
       allow read, write: if isAuthedTeacher();
 
+      match /documents/{docId} {
+        // portal-authenticated teachers can create valid documents
+        allow create: if isAuthedTeacher() && isValidDocumentRequest();
+        // teachers can only update their own documents and only if they're valid
+        allow update: if isAuthedTeacher() && isValidDocumentRequest() && resourceMatchUserId();
```

Comment:
I don't know if you need to worry about it. But in theory with these rules a teacher could take one of their own documents and change the uid to be that of another teacher.